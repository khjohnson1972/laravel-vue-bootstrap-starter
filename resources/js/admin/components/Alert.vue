<template>
    <div
        v-if="shown"
        class="alert rounded-0 mb-3 alert-dismissible fade show"
        role="alert"
        :class="status"
    >
        <span
            class="font-weight-bold mr-2"
        >{{ alert.title }}</span>
        <span>{{ alert.message }}</span>
        <button
            type="button"
            class="close"
            data-dismiss="alert"
            aria-label="Close"
        >
            <span aria-hidden="true">&times;</span>
        </button>
    </div>
</template>

<script>
export default {
  props: {
    alert: {
      type: Object,
      default: () => {}
    }
  },
  computed: {

    shown: function () {
      return this.alert.title || this.alert.message
    },

    status: function () {
      if (['primary', 'secondary', 'success', 'danger', 'warning', 'info', 'light', 'dark'].includes(this.alert.context)) {
        return 'alert-' + this.alert.context
      }

      return 'alert-info'
    }
  }
}
</script>
