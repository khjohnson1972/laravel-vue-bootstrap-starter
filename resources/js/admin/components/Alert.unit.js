import Alert from './Alert.vue'

describe('Alert.vue Component', () => {
  let wrapper = null

  // creates a new wrapper after each test
  beforeEach(() => {
    wrapper = shallowMount(Alert, {
      ...createComponentMocks(),
      props: {
        alert: {
          message: 'test'
        }
      }
    })
  })

  // unmount is now Vue3 -  the wrapper after each test
  afterEach(() => {
    wrapper.unmount()
  })

  test('exports a valid component', () => {
    expect(Alert).toBeAComponent()
  })

  test('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('test shown() returns message', () => {
    expect(wrapper.vm.shown).toBe('test')
  })

  test('test shown() returns title', async () => {
    await wrapper.setProps({
      alert: {
        title: 'title'
      }
    })
    expect(wrapper.vm.shown).toBe('title')
  })

  test('test status() returns class name', async () => {
    await wrapper.setProps({
      alert: {
        context: 'success'
      }
    })
    expect(wrapper.vm.status).toBe('alert-success')
  })
})
