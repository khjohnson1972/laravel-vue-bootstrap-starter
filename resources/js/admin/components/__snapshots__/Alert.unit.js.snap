// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert.vue Component matches snapshot 1`] = `<div class="alert rounded-0 mb-3 alert-dismissible fade show alert-info" role="alert"><span class="font-weight-bold mr-2"></span><span>test</span><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">Ã—</span></button></div>`;
